zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -420,6 +420,12 @@ impl<F: Field> ExecutionConfig<F> {
                     1.expr(),
                 )
             });
+            // For every step, is_create and is_root are boolean.
+            cb.require_boolean(
+                "step.is_create is boolean",
+                step_curr.state.is_create.expr(),
+            );
+            cb.require_boolean("step.is_root is boolean", step_curr.state.is_root.expr());
```

Comment:
We should only constrain they are boolean when `q_step` is enabled?

Reply:
I considered that. But then, even when `q_step` is disabled, those values will be `0`? So it still is boolean, hence kept it this way. Wdyt?

Reply:
ah but won't these cells in `StepState` be used for storing other non-boolean values while the `q_step` is disabled? I thought this is the reason that some tests are failing.